cdap/http.go

```diff
@@ -44,13 +44,21 @@ func httpCall(client *http.Client, req *http.Request) ([]byte, error) {
 
 	b, err := doHTTPCall(client, req)
 
-	// CDAP REST intermittently returns 500 internal errors, we will retry on 500s once.
+	// CDAP REST intermittently returns 500, 504, etc. internal errors, we will retry on 5xxs once.
 	var e *httpError
-	if errors.As(err, &e) && e.code == 500 {
-		log.Print("retrying on intermittent 500 error in 2 seconds")
-		time.Sleep(2 * time.Second)
-		b, err = doHTTPCall(client, req)
-	}
+    // poor man's 5x EBO.
+    s := 2
+    for i:=0; i < 5; i++ {
+        if errors.As(err, &e) && e.code >= 500 && e.code < 600 {
+            log.Printf("retrying on intermittent 5xx error in %v seconds", s)
+            time.Sleep(2 * time.Second)
+            s *= 2
+            b, err = doHTTPCall(client, req)
+            if err != nil {
```

Comment:
I think you meant s * time.Second?

My personal preference would be to just do a linear 5 or 10 second sleep 2 more times. I have hit that 504 occassionally to and the very next call usually passes so I don't think we need 5 retries.